src/main/java/org/opensearch/index/codec/customcodecs/QatDeflateCompressionMode.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.customcodecs;
+
+import org.apache.lucene.codecs.compressing.CompressionMode;
+import org.apache.lucene.codecs.compressing.Compressor;
+import org.apache.lucene.codecs.compressing.Decompressor;
+import org.apache.lucene.store.ByteBuffersDataInput;
+import org.apache.lucene.store.DataInput;
+import org.apache.lucene.store.DataOutput;
+import org.apache.lucene.util.ArrayUtil;
+import org.apache.lucene.util.BytesRef;
+
+import java.io.IOException;
+import java.util.function.Supplier;
+
+import com.intel.qat.QatZipper;
+
+/** QAT_DEFLATE ompression Mode */
+public class QatDeflateCompressionMode extends CompressionMode {
```

Comment:
The `QatDeflateCompressionMode` and `QatLz4CompressionMode` are looking identical to me (the only difference is `QatZipper qatZipper` instance that is algorithm specific), may be could have once class with 2 factory methods? Fe.:

```java
public class QatCompressionMode extends CompressionMode {
    static CompressionMode  defalte() {
        return new QatCompressionMode(QatZipperFactory.createInstance(
                QatZipper.Algorithm.DEFLATE,
                compressionLevel,
                mode,
                QatZipper.PollingMode.PERIODICAL
            ), ...);

         static CompressionMode lz4() {
                return new QatCompressionMode(QatZipperFactory.createInstance(
                      QatZipper.Algorithm.LZ4,    
                      compressionLevel, mode, 
                      QatZipper.PollingMode.PERIODICAL), ...);
    }    
}
```

Reply:
@reta Refactored. We just need to pass one additional parameter to `QatCompressionMode`. 

In the checks, I'm not sure why one of the backward compatibility tests failed.

Reply:
> In the checks, I'm not sure why one of the backward compatibility tests failed.

Thanks @mulugetam , I think the test is unstable :((((